word_segmentation.py

```diff
@@ -349,233 +413,225 @@ def preprocess_Burmese(demonstrate):
     Args:
-        read_filename: Address of the input file
-        write_filename: Address of the output file
+        char_brkpoints: break points for grapheme clusters
+        word_brkpoints: break points for words
     """
-    wfile = open(write_filename, 'w')
-    with open(read_filename) as f:
-        for line in f:
-            line = line.strip()
-            if len(line) == 0:
-                continue
-            new_line = ""
-            for i in range(len(line)):
-                ch = line[i]
-                if ch == " ":
-                # If you want to put lines bars around punctuations as well, you should use comment previous if and
-                # uncomment the next if.
-                # The following if will put bars for !? as |!||?| instead of |!|?|. This should be fixed if the
-                # following if is going to be used. It can easily be fixed by keeping track of the last character in
-                # new_line.
-                # if 32 <= ord(ch) <= 47 or 58 <= ord(ch) <= 64:
-                    if i == 0:
-                        if i+1 < len(line) and line[i+1] == "|":
-                            new_line = new_line + "|" + ch
-                        else:
-                            new_line = new_line + "|" + ch + "|"
-                    elif i == len(line)-1:
-                        if line[i-1] == "|":
-                            new_line = new_line + ch + "|"
-                        else:
-                            new_line = new_line + "|" + ch + "|"
-                    else:
-                        if line[i-1] != "|" and line[i+1] != "|":
-                            new_line = new_line + "|" + ch + "|"
-                        if line[i-1] == "|" and line[i+1] != "|":
-                            new_line = new_line + ch + "|"
-                        if line[i-1] != "|" and line[i+1] == "|":
-                            new_line = new_line + "|" + ch
-                        if line[i-1] == "|" and line[i+1] == "|":
-                            new_line = new_line + ch
-                else:
-                    new_line += ch
-            new_line += "\n"
-            wfile.write(new_line)
+    bies = np.zeros(shape=[4, len(char_brkpoints)-1])
+    word_ind = 0
+    for i in range(len(char_brkpoints)-1):
+        word_st = word_brkpoints[word_ind]
+        word_fn = word_brkpoints[word_ind + 1]
+        char_st = char_brkpoints[i]
+        char_fn = char_brkpoints[i+1]
+        if char_st == word_st and char_fn != word_fn:
+            bies[0, i] = 1
+            continue
+        if char_st != word_st and char_fn != word_fn:
+            bies[1, i] = 1
+            continue
+        if char_st != word_st and char_fn == word_fn:
+            bies[2, i] = 1
+            word_ind += 1
+            continue
+        if char_st == word_st and char_fn == word_fn:
+            bies[3, i] = 1
+            word_ind += 1
+            continue
+    return bies
```

Comment:
Consider making BIES a type, which could be a NamedTuple with only one element, a string.  You get better type safety that way.